pkg/integration/replication/synchronous_replication_test.go

```diff
@@ -125,21 +122,21 @@ func (suite *SyncTestSuite) TestPrecommitStateSync() {
 			defer wg.Done()
 			ctx, client, cleanup := suite.ClientForReplica(followerID)
 			defer cleanup()
-			for {
-				select {
-				case <-startCh:
-					suite.Run(fmt.Sprintf("test replica sync state %d", followerID), func() {
-						state, err := client.CurrentState(ctx)
-						require.NoError(suite.T(), err)
-						suite.Require().Equal(state.PrecommittedTxId, masterState.TxId)
-						suite.Require().Equal(state.PrecommittedTxHash, masterState.TxHash)
-					})
-					return
-				}
+			for range startCh {
```

Comment:
This would not run at all because I don't think range checks for nil

https://go.dev/play/p/pJ1sU5I9x6n